tests/route/test_route_flap.py

```diff
@@ -0,0 +1,159 @@
+import math
+import os
+import yaml
+import requests
+import time
+
+import json
+import logging
+import re
+import ipaddress
+import pytest
+import ptf.testutils as testutils
+import ptf.packet as scapy
+
+from ptf.mask import Mask
+
+LOOP_TIMES_LEVEL_MAP = {
+    'debug': 1,
+    'basic': 5,
+    'confident': 25,
+    'thorough': 50,
+    'diagnose': 100
+}
+
+logger = logging.getLogger(__name__)
+
+def change_route(operation, ptfip, neighbor, route, nexthop, port):
+    url = "http://%s:%d" % (ptfip, port)
+    data = {"command": "neighbor %s %s route %s next-hop %s" % (neighbor, operation, route, nexthop)}
+    r = requests.post(url, data=data)
+    assert r.status_code == 200
+
+def announce_route(ptfip, neighbor, route, nexthop, port):
+    change_route("announce", ptfip, neighbor, route, nexthop, port)
+
+
+def withdraw_route(ptfip, neighbor, route, nexthop, port):
+    change_route("withdraw", ptfip, neighbor, route, nexthop, port)
+
+
+def get_send_port(duthost, tbinfo, port_info):
+	dev_port = port_info[0][1]
+	mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+	logger.info("mg_facts: %s" % mg_facts)
+	member_port = mg_facts['minigraph_portchannels'][dev_port]['members']
+	logger.info("member_port: %s" % member_port)
+	send_port = mg_facts['minigraph_ptf_indices'][member_port[0]]
+	logger.info("src_port: %d" % send_port)
+	return send_port
+
+
+def send_recv_ping_packet(ptfadapter, src_port, dst_mac, dst_ip):
+	pkt = testutils.simple_icmp_packet(
+									eth_dst = dst_mac, 
+									ip_dst = dst_ip, 
+									icmp_type=8)
+
+	ext_pkt = pkt.copy()
+	ext_pkt['Ether'].src = dst_mac
+
+	masked_exp_pkt = Mask(ext_pkt)
+	masked_exp_pkt.set_do_not_care_scapy(scapy.Ether,"dst")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.IP,"ttl")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "tos")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "len")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "id")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+	masked_exp_pkt.set_do_not_care_scapy(scapy.ICMP, "chksum")
+
+	logger.info('send ping request packet source port id {} dmac: {} dip: {}'.format(src_port, dst_mac, dst_ip))
+	testutils.send(ptfadapter, src_port, pkt)
+	testutils.verify_packet(ptfadapter, masked_exp_pkt, src_port, timeout=None)
+
+
+def get_ip_route_info(duthost):
+	output = duthost.command("ip route")['stdout']
+	output.replace('\n', ' ').replace('\r', ' ')
+	return output
+
+
+def get_url_port(duthost, ptfhost, nexthop):
```

Comment:
Please use the method in tests\bgp\test_bgp_allow_list.py for the more reliable way of getting TCP port of exabgp.

Reply:
OK